docs/reference/contrib/searchpromotions.md

```diff
@@ -41,12 +41,25 @@ To retrieve a list of promoted search results for a particular search query, you
 
 <ul>
     {% for search_promotion in search_promotions %}
-        <li>
-            <a href="{% pageurl search_promotion.page %}">
-                <h2>{{ search_promotion.page.title }}</h2>
-                <p>{{ search_promotion.description }}</p>
-            </a>
-        </li>
+        {% if search_promotion.page %}
+            <li>
+                <a href="{% pageurl search_promotion.page %}">
+                    <h2>{{ search_promotion.page.title }}</h2>
+                    <p>{{ search_promotion.description }}</p>
+                </a>
+            </li>
+        {% else %}
+            <li>
+                <a href="{{ search_promotion.external_link }}">
+                    {% if search_promotion.external_title %}
+                        <h2>{{ search_promotion.external_title }}</h2>
+                    {% else %}
+                        <h2>{{ search_promotion.external_link }}</h2>
+                    {% endif %}
+                    <p>{{ search_promotion.description }}</p>
+                </a>
+            </li>
+        {% endif %}
     {% endfor %}
```

Comment:
Sorry. Must have missed this. Docs are done. Thanks. 

It may be nicer to put the if/else inside the li element and only output the description once.